### DISCUSSION TITLE
How can I add a cask that is not an app, dmg, or package?

### DISCUSSION THREAD

**Discussion by nik312123** [Homebrew/discussions#5442] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

I would like advice on Homebrew expectations for a cask in which the app is not self contained in just a .app file, does not have a dmg installation, and does not have a package installation. This example does not appear to be mentioned [here](https://docs.brew.sh/Adding-Software-to-Homebrew#casks).

In my specific instance, for example, the release can be obtained from GitHub as normal (e.g. https://github.com/[example-user]/[example-name]/releases/download/[example-release-filename].tar.gz).

Then, the contents of the directory created when extracting the archive (e.g. [example-release-filename]) are moved into a new folder in the /Applications directory (e.g. contents of [example-release-filename] moved to /Applications/[example-name]).

I would appreciate any advice regarding the preferred way to do this in Homebrew casks. Thank you!

**Comment by krehel** [comment, marked as answer]
Hi @nik312123 - if the extracted archive is a folder, you can use `suite` which will move the folder to `/Applications` - you can look at [Chemdoodle](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/c/chemdoodle.rb) as an example.

If the archive is files only, I believe you can take a look at [Chatty](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/c/chatty.rb) as an example which uses a preflight block to move files to a staging location first and then invokes `suite` to add to `/Applications`

Hope this helps.

**Reply by nik312123** [reply]
Chatty's case is exactly what I needed! Though, in my instance, I used `FileUtils.mv(...)` to rename the directory to remove the version information before moving it into `/Applications` (e.g. `FileUtils.mv(staged_path.join("[example-path-with-version-information]"), staged_path.join("[example]"))`).

**Reply by krehel** [reply]
Glad you were able get it working! 😄
